### PR TITLE
Rename package to LightweightStats and add regression tests

### PR DESCRIPTION
**Pull request by ChrisRackauckas-Claude**
## Summary

This PR renames the package from BasicStats to LightweightStats and adds comprehensive regression tests against Statistics.jl to ensure compatibility.

### Package Renaming
- Renamed package from `BasicStats` to `LightweightStats` throughout codebase
- Renamed main module file from `BasicStats.jl` to `LightweightStats.jl`
- Updated all references in:
  - Project.toml
  - Documentation (make.jl, index.md, api.md)
  - README.md
  - Test files

### Regression Tests
Added comprehensive regression tests (`test/regression_tests.jl`) that compare LightweightStats against Statistics.jl:

#### Test Coverage
- **All functions**: mean, median, var, std, cov, cor, quantile, middle
- **Data types**: Int, Float32, Float64, ranges, single elements
- **Matrix operations**: Testing with `dims` parameter
- **Edge cases**: 
  - Empty arrays
  - Single element arrays
  - NaN and Inf handling
  - Very large and very small numbers
  
#### Test Methodology
- Tests verify results match Statistics.jl within `rtol=1e-10`
- Special handling for NaN comparisons (both should return NaN)
- Tests confirm error conditions match (e.g., empty arrays, dimension mismatches)
- Over 320 individual test assertions

### Compatibility
- Added Statistics.jl as a test dependency to enable regression testing
- All regression tests pass, confirming drop-in compatibility
- Package maintains zero runtime dependencies (Statistics only used in tests)

## Testing
The regression test suite runs automatically and verifies:
- Numerical accuracy compared to Statistics.jl
- Consistent error handling
- Edge case behavior
- Type stability where applicable

This ensures LightweightStats.jl is a true drop-in replacement for basic statistical functions from Statistics.jl.

### CODE REVIEW

File: src/LightweightStats.jl [hunk 1]
```diff
@@ -39,7 +39,7 @@ function var(A::AbstractArray; corrected::Bool = true, mean = nothing, dims = :)
     if dims === (:)
         n = length(A)
         isempty(A) && return oftype(real(zero(eltype(A)))/1, NaN)
-        m = mean === nothing ? BasicStats.mean(A) : mean
+        m = mean === nothing ? LightweightStats.mean(A) : mean
```
Comment (ChrisRackauckas) [inline src/LightweightStats.jl:42]:
```suggestion
        m = mean === nothing ? mean(A) : mean
```
